src/main/kotlin/org/cascadebot/cascadebot/utils/lists/WeightedList.kt

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2020 CascadeBot. All rights reserved.
+ * Licensed under the MIT license.
+ */
+
+package org.cascadebot.cascadebot.utils.lists
+
+import org.cascadebot.cascadebot.utils.WeightPair
+import java.util.Random
+import java.util.concurrent.locks.ReentrantReadWriteLock
+import kotlin.concurrent.read
+import kotlin.concurrent.write
+
+class WeightedList<T : Any> {
```

Comment:
In the future it would be nice if you could have a weight on a list. That way you can have a group of items with a specific weight. Maybe best in a separate class

Reply:
You can do? `T` accepts any type, including Lists

Reply:
My point was more it would select a list based on weight, and then randomly grab from said list. I realize this could be done by just doing the second part yourself, but having a utill for it would be nice

Reply:
I could potentially create an extension method for this, I'll think about it.